### ISSUE TITLE
Some paging responses have inaccurate response types

### ISSUE THREAD

**Issue by RikkiGibson** [Azure/autorest.typescript#269]
e.g.
```ts
/**
 * Contains response data for the list operation.
 */
export type StorageAccountsListResponse = {
  /**
   * Gets the list of storage accounts and their properties.
   * **NOTE: This property will not be serialized. It can only be populated by the server.**
   */
  readonly value?: StorageAccount[];
  /**
   * The underlying HTTP response.
   */
  _response: msRest.HttpResponse & {
      /**
       * The response body as text (string format)
       */
      bodyAsText: string;
      /**
       * The response body as parsed JSON or XML
       */
      parsedBody: StorageAccountListResult;
    };
};

/**
 * @interface
 * An interface representing the StorageAccountListResult.
 * The response from the List Storage Accounts operation.
 *
 * @extends Array<StorageAccount>
 */
export interface StorageAccountListResult extends Array<StorageAccount> {
}
```

At runtime the storage accounts are assigned to `0`, `1`, `2`, etc. properties on the response object, when we should be creating a response array and assigning additional properties as needed.

**Comment by RikkiGibson** [comment]
https://github.com/Azure/autorest.typescript/blob/master/src/azure/Templates/PageModelTemplate.cshtml#L29 seems relevant

**Comment by RikkiGibson** [comment]
This didn't show up because the paging tests were using the callback version that just grabs the body

**Comment by RikkiGibson** [comment]
Solved by #270
